pkg/deployment/deployment.go

```diff
@@ -366,7 +396,17 @@ func ConditionalDeploy(
 
 	if status.Conditions.IsUnknown(readyCondition) {
 		log.Info(fmt.Sprintf("%s Unknown, starting %s", readyCondition, deployName))
-		err = deployFunc(ctx, helper, obj, sshKeySecret, inventoryConfigMap, aeeSpec)
+		switch true {
+		case strings.Contains(deployName, "Configure"):
+			err = serviceInterface.Configure(ctx, helper, obj, sshKeySecret, inventoryConfigMap, aeeSpec)
+		case strings.Contains(deployName, "Validate"):
+			err = serviceInterface.Validate(ctx, helper, obj, sshKeySecret, inventoryConfigMap, aeeSpec)
+		case strings.Contains(deployName, "Install"):
+			err = serviceInterface.Install(ctx, helper, obj, sshKeySecret, inventoryConfigMap, aeeSpec)
+		case strings.Contains(deployName, "Run"):
+			err = serviceInterface.Run(ctx, helper, obj, sshKeySecret, inventoryConfigMap, aeeSpec)
```

Comment:
We can probably change these to just be a `deployVerb` instead and use the `Install` `Configure` `Run` `Validate`. But that will likely change all of the logging, so we might want to address that in a different PR. It would simplify this though, we could just use case statements without looking for substrings.